### ISSUE TITLE
Form update infos should have a struct

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Risehat/freemedforms#269]
```
This issue was created by revision 9511a99b1354.

When e.g. FormManager calls FormIO.checkFormUpdates() and later
FormIO.updateForms(), it asks the user in between that there are forms
to update. As GUI/BusinessLogic separation this was moved into the
FormManager. But he has to get some information about what forms are
going to be updated and which data has changed.

There would be need of a struct/list with infos of
GenericUpdateInformation and XmlFormName.

Eric, do you have an idea here?
```

Original issue reported on code.google.com by `christian.a.reiter@gmail.com` on 6 Jan 2013 at 1:18

**Comment by GoogleCodeExporter** [comment]
```
Yes add

    virtual QString displayName() const = 0;

    virtual QString getHtmlUpdateMessagr() const = 0;

in IFormIO

then manage a view with a listview for the IO displaynames and a QTextBrowser 
for the update information
```

Original comment by `eric.mae...@gmail.com` on 8 Jan 2013 at 10:31

**Comment by GoogleCodeExporter** [comment]
```
Hm. I wanted to avoid passing a html text to the caller, because it is 
completely unflexible. This could be done additionally with a "QString 
getHtmlUpdateMessage()" method, but just additionally.

I always try to think of different ways to shot the information. Just think 
about having FMF on a tablet some time, or maybe selectively update just some 
Forms, others not (you somehow have to be able to check them PER FORM e.g.)
So I suppose it is necessary to give the update information in a structured way 
to the caller, and let him decide what he does with it, providing flexibility. 
Just a html text cuts of this flexibility.

So this info must contain a list of forms, and each forms a list of version 
updates, right?
That should not be so complicated - I could have designed a new struct. But I 
just wanted to ask you if you have a better idea, or maybe there is already 
such a data structure.
```

Original comment by `christian.a.reiter@gmail.com` on 9 Jan 2013 at 6:26

**Comment by GoogleCodeExporter** [comment]
```
Then use the Form::FormIODescription

    QList<Form::FormIODescription> checkForUpdates() = 0; // return the list of forms to update
    bool update(const QList<Form::FormIODescription> &forms); // update the following forms

So user can see what can be updated, and even, he can select the forms to 
update (using view/check...)

You want to code this ?
```

Original comment by `berenger...@gmail.com` on 14 Jan 2013 at 3:39

**Comment by GoogleCodeExporter** [comment]
```
Yes. Give me some time.
```

Original comment by `christian.a.reiter@gmail.com` on 15 Jan 2013 at 5:29
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision c9c3e6dfe0fd.
```

Original comment by `eric.mae...@gmail.com` on 8 Feb 2013 at 6:40
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 0f7c421a021d.
```

Original comment by `eric.mae...@gmail.com` on 8 Feb 2013 at 6:40
